### PR TITLE
[stable25] fix(user-management): Fix deleting user in hidden group(s)

### PR DESCRIPTION
**Pull request by szaimen**
Supersedes https://github.com/nextcloud/server/pull/36579

### CODE REVIEW

File: apps/settings/src/store/users.js [hunk 1]
```diff
@@ -184,6 +184,10 @@ const mutations = {
 				state.userCount-- // decrement Active Users count
 				user.groups.forEach(userGroup => {
 					const group = state.groups.find(groupSearch => groupSearch.id === userGroup)
+					if (!group) {
+						console.warn('User group ' + userGroup + ' does not exist during user removal')
+						return
+					}
```
Comment (artonge) [inline apps/settings/src/store/users.js:190]: I would map and filter instead of this

Comment (szaimen) [reply apps/settings/src/store/users.js:190]: @ChristophWurst could you have a look at the comment? Not sure if I can follow Louis' suggestion...

Comment (ChristophWurst) [reply apps/settings/src/store/users.js:190]: It gets the same result of `undefined`. This is a backport. Let's leave it like the original PR and refactoring is desired one can clean up the code on `master`

**Review by artonge — APPROVED** [review]

**Review by blizzz — APPROVED** [review]

**Pull request merged**